### DISCUSSION TITLE
Add empirical HMC

### DISCUSSION THREAD

**Discussion by rlouf** [blackjax-devs/blackjax#96] [New features]
https://arxiv.org/pdf/1810.04449.pdf

This requires:
1. A jittered version of HMC (as in #12);
2. A new tuning algorithm that returns an array of trajectory length;
3. Dual Averaging;
4. Mass matrix adaptation;
5. The NUTS kernel to use with (1).

(1) is trivial to
implement with BlackJAX and (3,4,5) are part of the NUTS sampler.

**Comment by sethaxen** [comment]
Some more discussion and comparisons here: https://discourse.mc-stan.org/t/hmc-jittered-vs-nuts-on-1000-dimensional-standard-normal/8383

**Comment by rlouf** [comment]
The Stan Discourse will never cease to amaze me.
